packages/ra-ui-materialui/src/input/ReferenceArrayInput.tsx

```diff
@@ -141,19 +143,33 @@ const ReferenceArrayInput: FC<ReferenceArrayInputProps> = ({
         input,
     });
 
+    const listContext = useMemo(
+        () => ({
+            ...controllerProps,
+            setSort: controllerProps.setSortForList,
+        }),
+        [controllerProps]
+    );
+
     const translate = useTranslate();
 
     return (
-        <ReferenceArrayInputView
-            id={id}
-            input={input}
-            isRequired={isRequired}
-            meta={meta}
-            translate={translate}
-            children={children}
-            {...props}
-            {...controllerProps}
-        />
+        <ResourceContextProvider value={props.reference}>
+            <ReferenceArrayInputContextProvider value={controllerProps}>
+                <ListContextProvider value={listContext}>
```

Comment:
I don't get it. What's the use of a ReferenceArrayInputContext if all you need is already in the ListContext?

Reply:
I agree it is mostly redundant now but it does provide the `warning` prop which indicates non "fatal" errors regarding references.